flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpillingAdaptiveSpanningRecordDeserializer.java

```diff
@@ -557,6 +579,22 @@ private void addNextChunkFromMemorySegment(MemorySegment segment, int offset, in
 			}
 		}
 
+		@Nullable
+		MemorySegment copyToTargetSegment() {
+			if (lengthBuffer.position() > 0) {
+				MemorySegment segment = MemorySegmentFactory.allocateUnpooledSegment(lengthBuffer.remaining());
+				segment.put(0, lengthBuffer, lengthBuffer.remaining());
+				return segment;
+			}
+
+			//TODO we skip the case of large record which size exceeds THRESHOLD_FOR_SPILLING, and it needs to consider reading from spillFile
+			if (recordLength != -1) {
+				return MemorySegmentFactory.wrap(buffer);
+			}
+
+			return null;
```

Comment:
in which case can this return `null`?

Reply:
If there is no remaining buffer in deserializer, it would return `null`.